src/server/logging/log_interceptor.js

```diff
@@ -0,0 +1,47 @@
+import Stream from 'stream';
+import { get, isEqual } from 'lodash';
+
+function doTagsMatch(event, tags) {
+  return isEqual(get(event, 'tags'), tags);
+}
+
+export class LogInterceptor extends Stream.Transform {
+  constructor() {
+    super({
+      readableObjectMode: true,
+      writableObjectMode: true
+    });
+  }
+
+  /**
+   *  Since the upgrade to hapi 14, any socket read
+   *  error is surfaced as a generic "client error"
+   *  but "ECONNRESET" specifically is not useful for the
+   *  logs unless you are trying to debug edge-case behaviors.
+   *
+   *  For that reason, we downgrade this from error to debug level
+   *
+   *  @param {object} - log event
+   */
+  downgradeIfEconnreset(event) {
+    const isClientError = doTagsMatch(event, ['error', 'client', 'connection']);
```

Comment:
Hapi is using the following order `['connection', 'client', 'error']` per https://github.com/hapijs/hapi/blob/master/lib/connection.js#L96